src/Collection/Collection.php

```diff
@@ -20,26 +20,17 @@ class_exists(Pagination::class);
  */
 class Collection implements IteratorAggregate, Countable
 {
-    private $elements;
-    private $count;
-    private $pagination;
-
-    /**
-     * @var Microservice|null
-     */
-    private $microservice;
+    private ?Microservice $microservice;
```

Comment:
```suggestion
    private ?Microservice $microservice = null;
```
So you can remove the set in the constructor